docs/monitor-etcd.md

```diff
@@ -0,0 +1,29 @@
+# Background
+
+In Pai version before 0.10.0, pai did not have a monitor solution for etcd deployed
+by OpenPai, so admin will not get notified when something wrong with etcd and can not
+running correctly.
+
+To monitor etcd, we have watchdog service, this service monitor pods with special
+labels and check its running status via readiness probe. If you have deployed
+kubernetes using version larger than 0.10.0, you can ignore this document, because
+the etcd has configured correctly. But if you have deployed kubernetes using version
+before 0.10.0, you should do following upgrade to configure etcd correctly so it can
+be monitored.
+
+# Upgrade steps
+
+Please note following steps involves restart etcd pods, the etcd process will stop for around 5 minutes, and **kubernetes can not work during the time**, please plan upgrade in advance to minimize the possible impact.
+
+1. Please use a [dev-box](pai-management/doc/how-to-setup-dev-box.md) to do following operations, as dev-box can provide a clean environment.
+2. Checkout the 0.10.0 version branch of pai: `cd $PAI_ROOT && git checkout v0.10.0`.
+3. Apply upgrade script: `cd $PAI_ROOT/src/etcd-upgrade/build && kubectl apply -f ds.yaml`.
```

Comment:
>ds.yaml [](start = 82, length = 7)

use a daemonSet to upgrade etcd is way too heavy. by the way, can we just label the pod? k8s does support attaching a label to a node on the fly, right?

Reply:
Yes, we are updating the etcd.yaml, by using DaemonSet to deploy update script in every node and update etcd.yaml.